miner/proposal_builder.go

```diff
@@ -68,23 +71,37 @@ type ProposalBuilder struct {
 	tortoise  votesEncoder
 	syncer    system.SyncStateProvider
 
+	mu      sync.Mutex
+	signers map[string]*signerSession
```

Comment:
Is it required to persist the session here? Wouldn't it suffice to persist `signing.EdSigner` in here and create a fresh session for each signer in the `build()` method?

Reply:
it will query database for various data repeatedly on every layer otherwise. it is better not to, if we can store a bit of data in memory